src/Microsoft.Maintainability.Analyzers/UnitTests/AvoidUnusedPrivateFieldsTests.cs

```diff
@@ -12,12 +13,74 @@ public class AvoidUnusedPrivateFieldsTests : DiagnosticAnalyzerTestBase
     {
         protected override DiagnosticAnalyzer GetBasicDiagnosticAnalyzer()
         {
-            return new BasicAvoidUnusedPrivateFieldsAnalyzer();
+            return new AvoidUnusedPrivateFieldsAnalyzer();
         }
 
         protected override DiagnosticAnalyzer GetCSharpDiagnosticAnalyzer()
         {
-            return new CSharpAvoidUnusedPrivateFieldsAnalyzer();
+            return new AvoidUnusedPrivateFieldsAnalyzer();
+        }
+
+        [Fact]
+        public void CSharp_CA1823DiagnosticCases()
+        {
+            VerifyCSharp(@"
+public class Class
+{
+    private string fileName = ""data.txt"";
+    private int Used1 = 10;
+    private int Used2;
+    private int Unused1 = 20;
+    private int Unused2;
+    public int Unused3;
+
+    public string FileName()
+    {
+        return fileName;
+    }
+
+    private int Value => Used1 + Used2;
+}
+",
+            GetCA1823CSharpResultAt(7, 17, "Unused1"),
+            GetCA1823CSharpResultAt(8, 17, "Unused2"));
+        }
+
+        [Fact]
+        public void VisualBasic_CA1823DiagnosticCases()
+        {
+            VerifyBasic(@"
+Public Class Class1
+	Private fileName As String
+    Private Used1 As Integer = 10
+    Private Used2 As Integer
+    Private Unused1 As Integer = 20
+    Private Unused2 As Integer
+    Public Unused3 As Integer
+
+	Public Function MyFileName() As String
+		Return filename
+	End Function
+
+	Public Function MyValue() As Integer
+		Return Used1 + Used2
+	End Function
+End Class
+",
+            GetCA1823BasicResultAt(6, 13, "Unused1"),
+            GetCA1823BasicResultAt(7, 13, "Unused2"));
+        }
+
+        internal static string CA1823Name = "CA1823";
```

Comment:
I suggest using `AvoidUnusedPrivateFieldsAnalyzer .RuleId` rather than repeating the name here.